packages/api/src/chats/usecases/add-message-to-chat.usecase.ts

```diff
@@ -1,19 +1,23 @@
+import { MemoryService } from '@/ai/services/memory.service';
 import { AppConfigService } from '@/app-config/app-config.service';
 import { ChatsRepository } from '@/chats/chats.repository';
 import { AddMessageToChatRequestDto } from '@/chats/dtos/add-message-to-chat.request.dto';
 import { ChatMessageResponseDto } from '@/chats/dtos/chat-message.response.dto';
 import { ChatMessageMustHaveSenderException } from '@/chats/exceptions/chat-message-must-have-sender.exception';
 import { ChatNotFoundException } from '@/chats/exceptions/chat-not-found.exception';
+import { RedisChatMemoryNotFoundException } from '@/chats/exceptions/redis-chat-memory-not-found.exception';
 import { InternalServerErrorException } from '@/common/exceptions/internal-server-error.exception';
 import { Usecase } from '@/common/types/usecase';
 import { ChatMessageResponseSchema } from '@/contract/chats/chat-message.response.dto';
 import { Injectable } from '@nestjs/common';
+import { encode } from 'gpt-3-encoder';
 
 @Injectable()
 export class AddMessageToChatUsecase implements Usecase {
   constructor(
     private readonly chatsRepository: ChatsRepository,
-    private readonly appConfigService: AppConfigService
+    private readonly appConfigService: AppConfigService,
+    private readonly memoryService: MemoryService
```

Comment:
could we do the memory consulting from the `ai.service` facade? I really wanted the `AiService` to be the only API Facade exposed. And this can be as simple as a properly named "getter" in the facade